pkg/vsphere/tasks/waiter.go

```diff
@@ -78,7 +85,44 @@ func WaitForResult(ctx context.Context, f func(context.Context) (ResultWaiter, e
 		}
 
 		log.Errorf(cerr.Error())
-		return nil, cerr
+		return info, cerr
 	}
 	return info, nil
 }
+
+func Retry(ctx context.Context, f func(context.Context) (ResultWaiter, error)) (*types.TaskInfo, error) {
+	r := rand.New(rand.NewSource(time.Now().UnixNano())) //creates a more unique random
+	var err error
+	var taskInfo *types.TaskInfo
+	backoffFactor := int64(1)
+
+	for {
+		taskInfo, err = WaitForResult(ctx, f)
+		if err != nil && taskInfo == nil {
```

Comment:
I am pretty sure the logic here can be simplified to `if taskInfo == nil` thoughts?